### DISCUSSION TITLE
MS SQL Link Not Fetching Tables

### DISCUSSION THREAD

**Discussion by bmc8923** [Budibase/budibase#11187] [Help] [answered]
Hi,
When I'm using the cloud Budibase. When i connect to my ms sql server it will connect successfully and show all the names of tables i can fetch. When i fetch these tables it will load for a while, then say "Tables Fetched Successfully." However no tables are fetched. Any ideas on why this might happen.

**Comment by luisdgeneMT** [comment, marked as answer]
I experienced a similar issue and noticed I had not properly set my Primary Keys. Once I fixed that it worked. Not sure if that would help in your case but it would not hurt to check.
